space-invaders.asm

```diff
@@ -7,6 +7,27 @@
 %define INVADERS_SHOOT_CYCLES 4
 %define BULLETS_MOVE_CYCLE 28
 
+%define INVALID_STATE 0x0000
+
+%define PLAYER_WIN 0
+%define INVADERS_WIN 1
+
+%define START_STATE 0
+%define GAME_STATE 1
+%define END_STATE 2
+
+%define START_KEY ' '
+%define RETRY_KEY 'r'
+
+%define MOVE_UP 0
+%define MOVE_RIGHT 1
+%define MOVE_DOWN 2
+%define MOVE_LEFT 3
+
+%define INVADERS_NEXT_OFFSET 2
```

Comment:
Similar to the bullet struct, this should probably also be done with the invaders struct.

Reply:
Nice!
I will fix it soon.

For the above, I will add some comments.

Thank you(^^)